language/move-vm/runtime/src/interpreter.rs

```diff
@@ -1253,6 +1499,11 @@ impl Frame {
                     Bytecode::MoveToGeneric(si_idx) => {
                         let resource = interpreter.operand_stack.pop()?;
                         let signer_reference = interpreter.operand_stack.pop_as::<StructRef>()?;
+
+                        if self.runtime_config.paranoid_type_checks {
+                            signer_reference.check_inner_type(&Type::Signer)?;
+                        }
+
```

Comment:
I believe this also wants to check that the value being moved is not a hot potato.

Reply:
We check this when we generate the writeset, where we can do a full check on the tree we want to write: 

i.e: the root of the tree has to have key capability and all the nested structs have store. I believe it's best for us to perform the checks there because inner nodes of the tree could potentially be mutated by write ref?